packages/components/addon/components/hds/table/th-selectable.js

```diff
@@ -0,0 +1,75 @@
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: MPL-2.0
+ */
+
+import Component from '@glimmer/component';
+import { action } from '@ember/object';
+import { guidFor } from '@ember/object/internals';
+import { tracked } from '@glimmer/tracking';
+
+export default class HdsTableThSelectableComponent extends Component {
+  @tracked isSelected = this.args.isSelected;
+
+  /**
+   * Generate a unique ID for the Checkbox
+   * @return {string}
+   */
+  checkboxId = 'checkbox-' + guidFor(this);
+
+  get ariaLabel() {
+    let { selectionAriaLabelSuffix } = this.args;
+    const prefix = this.isSelected ? 'Deselect' : 'Select';
+    if (selectionAriaLabelSuffix) {
+      return `${prefix} ${selectionAriaLabelSuffix}`;
+    } else {
+      return prefix;
+    }
+  }
+
+  @action
+  didInsert(checkbox) {
+    let { didInsert } = this.args;
+    if (typeof didInsert === 'function') {
+      didInsert(checkbox, this.args.selectionKey);
+      // we need to use a custom event listener here because changing the `checked` value via JS
+      // (and this happens with the "select all") doesn't trigger the `change` event
+      // and consequently the `aria-label` won't be automatically updated (and so we have to force it)
+      checkbox.addEventListener(
+        'toggle',
```

Comment:
as discussed, I suggest we prefix these events if we go down this path

Reply:
Opened PR https://github.com/hashicorp/design-system/pull/1912to address this, so we can discuss pros/cons of the proposed alternative approach.

Reply:
@alex-ju @KristinLBradley after further analysis (see comment https://github.com/hashicorp/design-system/pull/1912#issuecomment-1917085102) I have realized that the proposed solution was fundamentally flawed/incomplete, and have it working as expected would require a not-so-great change to the current code implementation.

I don't see many alternatives at this point:
- leave things as they are now, with the custom event
- have the `table/index.js` do DOM manipulation and update the `aria-label` values behind the scenes (this may become complex anyway, because the aria label suffix is declared at row level)
- imagine something even more complex (services, pubsub, etc)

I would lean towards leaving the code as it is (unless we decide to ask for help from other Ember devs in the company). 

Thoughts?